beatrix/src/test/java/org/killbill/billing/beatrix/integration/TestMigrationSubscriptions.java

```diff
@@ -237,8 +237,8 @@ public void testSimpleMigrationBundle() throws Exception {
         accountChecker.checkAccount(account.getId(), accountData, callContext);
 
         // We set both entitlement and billing date with desired value
-        final LocalDate entitlementMigrationDate = new LocalDate(2015, 12, 20);
-        final LocalDate billingMigrationDate = new LocalDate(2016, 2, 1);
+        final DateTime entitlementMigrationDate = new DateTime(2015, 12, 20, 0, 0, 0);
```

Comment:
I would argue, that it works, but it's more about what we want the test to do: Right now everything is aligned at `00:00:00`, (e.g the `account#reference_time` is and then we also force the subscriptions dates to be also aligned to `00:00:00`.  It seems fine with me, but as always we could try bother combinations if we wanted, but @reshmabidikar Already write some good tests to verify the dates being passed. @pierre Did you have something else in mind?

Reply:
My point was that the code implicitly assumes the reference time is midnight. I was thinking constructing the `DateTime` time portion using the reference time, or at least adding a comment.

Reply:
I've modified this now to use:

```
final DateTime entitlementMigrationDateBundle1 = internalCallContext.toUTCDateTime(new LocalDate(2015, 12, 20));
```

LMK if this makes sense. If so, there might be some other tests that need to be updated as above.

Reply:
Sounds good -- we can always fix the other tests next time we update them.

Just FYI, your fix works because of this: https://github.com/killbill/killbill/blob/793b841c0f5b7642df680d1d3bb50b9744942cb5/beatrix/src/test/java/org/killbill/billing/beatrix/integration/TestIntegrationBase.java#L513